### PR TITLE
Chunk Debug Renderer Fix

### PR DESCRIPTION
**Pull request by babbaj**
Makes the chunk border renderer(f3+g) render relative to the player's Y value

https://www.youtube.com/watch?v=eiOjrmH8XdM

### CODE REVIEW

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+@Mixin(DebugRendererChunkBorder.class)
+public class MixinDebugRenderChunkBorder {
+
+    /**
+     * Change chunk border renderer to work at any Y value
+     * @author Babbaj
+     */
+    @Overwrite
+    public void render(float partialTicks, long finishTimeNano) {
+        EntityPlayer player = FMLClientHandler.instance().getClientPlayerEntity();
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java]: This should only be enabled for cubic chunks worlds. Inject at HEAD, and cancel if it's cubic chunks world.

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        Tessellator tessellator = Tessellator.getInstance();
+        BufferBuilder bufferbuilder = tessellator.getBuffer();
+        double d0 = player.lastTickPosX + (player.posX - player.lastTickPosX) * (double)partialTicks;
+        double d1 = player.lastTickPosY + (player.posY - player.lastTickPosY) * (double)partialTicks;
+        double d2 = player.lastTickPosZ + (player.posZ - player.lastTickPosZ) * (double)partialTicks;
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java]: convert them to human names. Also mark the places that you changed

**Review by Niko-sk2x — CHANGES_REQUESTED** [review]

**Comment by babbaj** [comment]
did both

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        }
+
+        tessellator.draw();
+        GlStateManager.glLineWidth(2.0F);
+        bufferbuilder.begin(3, DefaultVertexFormats.POSITION_COLOR);
+
+        // Blue vertical lines
+        for (int j1 = 0; j1 <= 16; j1 += 16)
+        {
+            for (int l1 = 0; l1 <= 16; l1 += 16)
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java]: More variable names. And formatting. Use the supplied IDEA or eclipse formatter.

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                bufferbuilder.pos(chunkX + (double)j1, maxY, chunkZ + (double)l1).color(0.25F, 0.25F, 1.0F, 1.0F).endVertex();
+                bufferbuilder.pos(chunkX + (double)j1, maxY, chunkZ + (double)l1).color(0.25F, 0.25F, 1.0F, 0.0F).endVertex();
+            }
+        }
+
+        // Blue horizontal lines
+        //start from the offset
+        for (int k1 = (int)yOffset; k1 <= 256+(int)yOffset; k1 += 16)
+        {
+            double d8 = (double)k1 - playerY;
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinDebugRenderChunkBorder.java]: And here

**Comment by babbaj** [comment]
formatting??

**Comment by Niko-sk2x** [comment]
Curly braces on the same line, not next line (and probably a few other things will be changed if you use [the supplied formatter](https://github.com/OpenCubicChunks/CubicChunks/tree/MC_1.12/code-formatters/intellij-idea-formatter))

**Comment by babbaj** [comment]
ran intellij's formatter on it

**Comment by Niko-sk2x** [comment]
The variable names aren't changed yet, but now it doesn't hurt readability in significant way so I will merge anyway.

**Review by Niko-sk2x — APPROVED** [review]

**Pull request merged**